io.openems.edge.meter.siemens.pac1600/src/io/openems/edge/meter/siemens/pac1600/MeterSiemensPAC1600Impl.java

```diff
@@ -0,0 +1,166 @@
+import io.openems.edge.bridge.modbus.api.task.FC3ReadRegistersTask;
+import io.openems.edge.common.channel.value.Value;
+import io.openems.edge.common.component.OpenemsComponent;
+import io.openems.edge.common.modbusslave.ModbusSlave;
+import io.openems.edge.common.modbusslave.ModbusSlaveTable;
+import io.openems.edge.common.taskmanager.Priority;
+import io.openems.edge.common.type.TypeUtils;
+import io.openems.edge.meter.api.ElectricityMeter;
+import io.openems.edge.meter.api.MeterType;
+
+@Designate(ocd = Config.class, factory = true)
+@Component(//
+		name = "Meter.SiemensPAC1600", //
+		immediate = true, //
+		configurationPolicy = ConfigurationPolicy.REQUIRE //
+)
+public class MeterSiemensPAC1600Impl extends AbstractOpenemsModbusComponent
+		implements MeterSiemensPAC1600, ElectricityMeter, ModbusComponent, OpenemsComponent, ModbusSlave {
+
+	private MeterType meterType = MeterType.GRID;
+
+	@Reference
+	protected ConfigurationAdmin cm;
+
+
+	public MeterSiemensPAC1600Impl() {
+		super(//
+				OpenemsComponent.ChannelId.values(), //
+				ModbusComponent.ChannelId.values(), //
+				ElectricityMeter.ChannelId.values(), //
+				MeterSiemensPAC1600.ChannelId.values() //
+		);
+	}
+
+	@Reference(policy = ReferencePolicy.STATIC, policyOption = ReferencePolicyOption.GREEDY, cardinality = ReferenceCardinality.MANDATORY)
+	protected void setModbus(BridgeModbus modbus) {
+		super.setModbus(modbus);
+	}
+
+	@Activate
+	void activate(ComponentContext context, Config config) throws OpenemsException {
+		this.meterType = config.type();
+
+		if (super.activate(context, config.id(), config.alias(), config.enabled(), config.modbusUnitId(), this.cm,
+				"Modbus", config.modbus_id())) {
+			return;
+		}
+		// this.config = config;
+	}
+
+	@Deactivate
+	protected void deactivate() {
+		super.deactivate();
+	}
+
+	@Override
+	public MeterType getMeterType() {
+		return this.meterType;
+		// return this.config.type();
+	}
+
+	@Override
+	protected ModbusProtocol defineModbusProtocol() throws OpenemsException {
+		ModbusProtocol modbusProtocol = new ModbusProtocol(this, //
+				new FC3ReadRegistersTask(1, Priority.HIGH, //
+					m(ElectricityMeter.ChannelId.VOLTAGE_L1, new UnsignedDoublewordElement(1),ElementToChannelConverter.SCALE_FACTOR_1),
+					m(ElectricityMeter.ChannelId.VOLTAGE_L2, new UnsignedDoublewordElement(3),ElementToChannelConverter.SCALE_FACTOR_1),
+					m(ElectricityMeter.ChannelId.VOLTAGE_L3, new UnsignedDoublewordElement(5),ElementToChannelConverter.SCALE_FACTOR_1),
+					
+					m(ElectricityMeter.ChannelId.CURRENT_L1, new UnsignedDoublewordElement(7),ElementToChannelConverter.SCALE_FACTOR_MINUS_1),
+					m(ElectricityMeter.ChannelId.CURRENT_L2, new UnsignedDoublewordElement(9),ElementToChannelConverter.SCALE_FACTOR_MINUS_1),
+					m(ElectricityMeter.ChannelId.CURRENT_L3, new UnsignedDoublewordElement(11),ElementToChannelConverter.SCALE_FACTOR_MINUS_1),
+					
+					new DummyRegisterElement(13, 18), //
```

Comment:
If you like, you could still read those elements and create channels like `VOLTAGE_L1_L2` in `MeterSiemensPAC1600.ChannelId`. Same for the other DummyRegisters.